tern/analyze/default/command_lib/command_lib.py

```diff
@@ -159,6 +159,10 @@ def set_command_attrs(command_obj):
 def collate_snippets(snippet_list, package=''):
     '''Given a list of snippets, make a concatenated string with all the
     commands'''
+    # Escape any braces that might confuse Python formatting
+    for i, snip in enumerate(snippet_list):
+        if '{}' in snip:
+            snippet_list[i] = snip.replace('{}', '{{}}')
```

Comment:
How about something like this? It's more generic and will catch scripts that contain things like `${env_to_eval}`:
```
 replace_dict = {r'{': r'{{',  r'}': r'}}'}                                              
 for i, snip in enumerate(snippet_list):
        for key in replace_dict.keys():                                
               snippet_list[i] = re.sub(key, replace_dict[key], snippet) 
```

Reply:
Tern doesn't have an issue with `{<text>}` or `${<text>}` (no ValueError thrown in these cases) -- the issue is only with empty brackets so I tried to use a solution that avoided using `re` . I tested my change with the following Dockerfile with no errors and a report was generated. I can try your more comprehensive solution, though.

```
FROM ubuntu:20.04
RUN apt-get update && \
    apt-get install -y python3 python3-pip
RUN pip3 install --upgrade pip==9.0.3
COPY patch patch
# Patch some stuffs. Patch name MUST be something like 01-pip-desc
RUN for i in `ls patch`; \
    do module_name=$(echo $i | awk -F '-' '{print $2}'); \
    patch -d $(pip3 show $module_name | grep Location | awk '{print $2}') -p1 < patch/$i; \
    done
RUN rm -r patch
ENV TEST_ENDPOINT=https://test.myapi.com/v1/AUTH_${env_to_eval}
CMD ["/bin/sh"]
```

Reply:
The dockerfile parser should take care of escaping the curly braces. Maybe it missed `{}` kinds of string. I am ok with the change, but maybe it ought to be in the parser so we have parser related code in one place?

Reply:
I'm really confused :sweat_smile: I think I now understand the problem is specifically in formatting the snippet list with environment variables from the Dockerfile. Alright, I approve.